src/arch/riscv/config.cmake

```diff
@@ -27,6 +27,11 @@ config_option(
     DEPENDS "KernelArchRiscV"
 )
 
+# RISC-V currently has no instruction to count leading/trailing zeros,
+# so we provide verified library implementations.
+config_set(KernelClzNoBuiltin CLZ_NO_BUILTIN ON)
+config_set(KernelCtzNoBuiltin CTZ_NO_BUILTIN ON)
```

Comment:
@kent-mcleod I've put this in `riscv/config.cmake`, but it occurs to me that this might not be the best way to do it. Should I instead put this in each of the RISC-V platform configs? That way, when we eventually get a RISC-V platform that implements the forthcoming bitmanip extension, we won't accidentally leave it using the library functions.

Reply:
I expect that eventually some config automation will be added that sets these settings based off the micro-arch/extensions that the platform declares. So anything is fine until then IMO.